app/src/main/java/net/gsantner/markor/format/FormatRegistry.java

```diff
@@ -74,6 +75,32 @@ public class FormatRegistry {
     public final static OrgmodeTextConverter CONVERTER_ORGMODE = new OrgmodeTextConverter();
 
 
+
+    public static class Format {
+        public final @StringRes int format, name;
+        public final String ext;
+        public final TextConverterBase converter;
+
+        Format(int format, int name, String ext, final TextConverterBase converter) {
+            this.format = format;
+            this.name = name;
+            this.ext = ext;
+            this.converter = converter;
+        }
+    }
+
+    public static final Format[] FORMATS = new Format[] {
+            new Format(FormatRegistry.FORMAT_MARKDOWN, R.string.markdown, ".md", CONVERTER_MARKDOWN),
+            new Format(FormatRegistry.FORMAT_PLAIN, R.string.plaintext, ".txt", CONVERTER_PLAINTEXT),
+            new Format(FormatRegistry.FORMAT_TODOTXT, R.string.todo_txt, ".todo.txt", CONVERTER_TODOTXT),
+            new Format(FormatRegistry.FORMAT_WIKITEXT, R.string.wikitext, ".txt", CONVERTER_WIKITEXT),
```

Comment:
Is the matching now done pure by this? As far I can see it's still detectable by the isFileOutOfFormat function which can have own logic.

However it seems to be the pre-filter based on extension now, right? -> It's missing way too many things.
Overall it might be nice to have a central place to define formats, but at the other hand need to admit I find it better if each format defines their stuff on their own in their code.

Reply:
Matching is not done by this at all. That mechanism is unchanged. This list is just associating the following in one place:

- FormatRegistry type
- Resource string id
- canonical extension
- Converter

so that code working with multiple types can just reference this list instead of having these things associated ad-hoc in various places.